pkg/chunk/cache/memcached.go

```diff
@@ -89,7 +89,14 @@ func NewMemcached(cfg MemcachedConfig, client MemcachedClient, name string, reg
 						batchID: input.batchID,
 					}
 					res.found, res.bufs, res.missed = c.fetch(input.ctx, input.keys)
-					input.resultCh <- res
+					// No-one will be reading from resultCh if we were asked to quit
+					// during the fetch, so check again before writing to it.
+					select {
+					case <-c.quit:
+						return
+					default:
+						input.resultCh <- res
```

Comment:
sending on the channel inside the default case seems that it could still block if it enters the default clause just before quit is closed.
Can we make it a case instead?